apis/bigqueryconnection/v1alpha1/bigqueryconnectionconnection_types.go

```diff
@@ -37,9 +37,8 @@ type Parent struct {
 type BigQueryConnectionConnectionSpec struct {
 	Parent `json:",inline"`
 
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
```

Comment:
I make this optional in the CRD and add the validate in the KCC controller, so that we can validate the uuid format and give more user friendly message.